src/Dapr.AspNetCore/StateEntryModelBinder.cs

```diff
@@ -71,13 +71,20 @@ public async Task BindModelAsync(ModelBindingContext bindingContext)
             }
 
             var obj = await this.thunk(daprClient, this.storeName, key);
-            bindingContext.Result = ModelBindingResult.Success(obj);
 
-            bindingContext.ValidationState.Add(bindingContext.Result.Model, new ValidationStateEntry()
+            if (obj == null)
             {
-                // Don't do validation since the data came from a trusted source.
-                SuppressValidation = true,
-            });
+                bindingContext.Result = ModelBindingResult.Failed();
+            }
+            else
+            {
+                bindingContext.Result = ModelBindingResult.Success(obj);
+                bindingContext.ValidationState.Add(bindingContext.Result.Model, new ValidationStateEntry()
+                {
+                    // Don't do validation since the data came from a trusted source.
+                    SuppressValidation = true,
+                });
+            }
```

Comment:
We should add integration tests for this using https://github.com/dapr/dotnet-sdk/blob/master/test/Dapr.AspNetCore.IntegrationTest.App/DaprController.cs

I'd expect the following to happen when the state isn't found in the state store:
- If the parameter type is `StateEntry<Widget>` the `StateEntry<>` is non-null, but `StateEntry<>.Value` is null
- If the parameter type is `Widget` then the argument should be null.

For both of these cases, check `ModelState.IsValid` (should return true)

Reply:
I expect that this is already the behavior but the tests will help a lot.

Reply:
It would also really help to add some comments here about the expected behavior from a user POV (basically what I said above). The comments are useful because the relationship between this code and the outcome a user sees isn't obvious just from reading the code.